my-app/src/app/courses-page/courses-page.component.html

```diff
@@ -24,9 +24,15 @@
 </button>
 
 <app-courses-list
-    [courses]="courses"
+    [courses]="filteredCourses"
     (delete)="onDeleteCourse($event)"
     (edit)="onEditCourse($event)"
     (loadmoreCourse)="onLoadMoreCourses()"
+    *ngIf="isCoursesAvailable(); else noDataMessage"
```

Comment:
nit: usually the getter is defined instead of the function.
```typescript
*ngIf="isCoursesAvailable"

get isCoursesAvailable() {
  return !!!!this.filteredCourses.length;
}

Reply:
Done!